pkg/clusteragent/custommetrics/store.go

```diff
@@ -0,0 +1,137 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build kubeapiserver
+
+package custommetrics
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// Store is an interface for persistent storage of custom and external metrics.
+type Store interface {
+	SetExternalMetric(ExternalMetricValue) error
+	DeleteExternalMetric(hpaNamespace, hpaName, metricName string) error
+	ListAllExternalMetrics() ([]ExternalMetricValue, error)
```

Comment:
I think it'd be better to rename this to `ListAllExternalMetricsFromStore` or something that would highlight that this method is not the one used by the HPA Controller to get all the EM (will also help if people run CMD+F in the repo)

Reply:
`ListAllExternalMetricsFromStore` seems overly verbose and there would only be 2 functions in the repo with the same name for CMD+F?

Reply:
Up to you, I think the two methods are different and should be named accordingly. `ListAllExternalMetricsFromStore` was just a suggestion but it could also be `ListStoredExternalMetrics` ?

Reply:
@CharlyF reusing `store` in an method of the store interface seems a bit redundant, doc should be here to fix the disambiguation you're mentionning